src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceInfo.cs

```diff
@@ -192,8 +192,8 @@ private static bool TryGetGuid(INamedTypeSymbol interfaceSymbol, InterfaceDeclar
 
         public override int GetHashCode()
         {
-            // ContainingSyntax and ContainingSyntaxContext do not implement GetHashCode
-            return HashCode.Combine(Type, TypeDefinitionContext, InterfaceId);
+            // ContainingSyntax does not implement GetHashCode
+            return HashCode.Combine(Type, ThisInterfaceKey, BaseInterfaceKey, TypeDefinitionContext, InterfaceId);
```

Comment:
Maybe I'm missing something crucial here, but how does hashcode resolve to shared roslyn code?

Reply:
[HashCode.Combine](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/HashCode.cs) is a wrapper around [Hash.Combine](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/Roslyn/Hash.cs)

Reply:
Ah, one of the ugly hacks, srcGens in this repo can use to make the code more readable. Everyone else trying this will run into the .NET Standard 2.0 brick wall.